src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -171,6 +171,19 @@ public void RemoveEntries()
                 }
             }
 
+            public bool TryUseCachedEntries(out ImmutableArray<T> entries)
+            {
+                if (TryUseCachedEntries())
+                {
+                    entries = _states[_states.Count - 1].ToImmutableArray();
+                    return true;
+                }
```

Comment:
Missing an `else` here

Reply:
Lol I had the same bug in a previous PR that chuck pointed out, and somehow merged it over without the fix. Clearly gonna haunt me :p